### ISSUE TITLE
New release? 0.3.1 is a bit outdated now

### ISSUE THREAD

**Issue by samuela** [zargony/fuse-rs#147]
It looks like the latest version on [cargo](https://crates.io/crates/fuse) is 0.3.1, but that version still depends on an old version of time (pre-0.2.0). It's also 3 years old now. Would it be possible to get a new cargo release put out?

**Comment by javawolfpack** [comment]
Instructions here on GitHub say to use fuse = "0.4" in dependencies but it's not on cargo so can't use. Can we remedy this as assuming the fix for Timespec no longer being supported won't happen until this update happens?

**Comment by riaqn** [comment]
Is this package deprecated? I guess I have to use python instead...

**Comment by boulos** [comment]
It's likely that the author just forgot to do it in November. However, if you'd like, you can depend on this repo directly (or a rev=) by updating your Cargo.toml:

   ```fuse = { git = "https://github.com/zargony/fuse-rs" }```

and you could add rev=42e29d964e2b24e32e21b179b77e4d0d0a0857ac to get the current commit.

See https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories for more info.

**Comment by samuela** [comment]
Yeah, it'd still be nice to get a new official release out though. I'm starting to worry the project might be abandoned or something.

**Comment by zargony** [comment]
Sorry that this became so stale. I don't have intentions to abandon this project. My main job and personal stuff just keeps me so busy that I don't find much time to look into it. The plan actually was to get 0.4 (modernize branch) out next since it introduces several changes anyway. I agree the current state of 0.3 is unsatisfying and should be updated. But upgrading time would be a breaking change and require to bump to 0.4 and the modernize branch would have to become 0.5. On the other hand, the modernize branch will probably stay somewhat experimental for some time and it might be a good idea to have the current release updated first.

It'd be helpful if someone could prepare a PR based on the 0.3 tag which updates the desired dependencies and that can be release as 0.4.

**Comment by samuela** [comment]
Thanks for all your hard work on this @zargony! We really appreciate it!

**Comment by javawolfpack** [comment]
@boulos thanks, learned something new about cargo :) Would be nice if I had more time to really dive into Rust development, stumbled upon this library being curious if I could change an old operating systems assignment for FUSE in C into one in Rust for the graduate OS course I'm teaching this fall. 

Looking a bit more promising since I can now compile with that direct git/revision cargo statement, but if I can't get it working in the next few weeks won't be bothering with it. If I get some extra time might see if I can help contribute.

**Comment by cberner** [comment]
I have a fork which I've been developing. I released 0.4.0 from my fork, since I needed several of the new features since 0.3.1. Feel free to use it, until the official 0.4.0 is released. https://crates.io/crates/fuser
